librz/core/cmd_type.c

```diff
@@ -1627,78 +1693,16 @@ RZ_IPI int rz_cmd_type(void *data, const char *input) {
 			break;
 		}
 		break;
-	case 't': {
+	case 't': { // "tt"
 		if (!input[1] || input[1] == 'j') {
-			PJ *pj = NULL;
-			if (input[1] == 'j') {
-				pj = pj_new();
-				pj_o(pj);
-			}
-			char *name = NULL;
-			SdbKv *kv;
-			SdbListIter *iter;
-			SdbList *l = sdb_foreach_list(TDB, true);
-			ls_foreach (l, iter, kv) {
-				if (!strcmp(sdbkv_value(kv), "typedef")) {
-					if (!name || strcmp(sdbkv_value(kv), name)) {
-						free(name);
-						name = strdup(sdbkv_key(kv));
-						if (!input[1]) {
-							rz_cons_println(name);
-						} else {
-							const char *q = sdb_fmt("typedef.%s", name);
-							const char *res = sdb_const_get(TDB, q, 0);
-							pj_ks(pj, name, res);
-						}
-					}
-				}
-			}
-			if (input[1] == 'j') {
-				pj_end(pj);
-			}
-			if (pj) {
-				rz_cons_printf("%s\n", pj_string(pj));
-				pj_free(pj);
-			}
-			free(name);
-			ls_free(l);
+			rz_core_list_loaded_typedefs(core, input, TDB);
```

Comment:
this API is not good. We should avoid creating API that requires a  general "input" and keep parsing the input. The input should be parsed by the command handler and converted to the more appropriate type, then a more generic API should be called. For example, use RzOutputMode to indicate whether JSON or not.

Reply:
Right, missed that.

Reply:
Right. 
I have looked at another particular API and modified it accordingly to use `RzOutputMode` for JSON and all. 
Please see if that's what you had meant.
Thanks for the catching that, by the way. Learned new things while doing that!